File: core/core-domain/src/main/java/com/ssafy/witch/snack/Snack.java
package com.ssafy.witch.snack;

import java.time.LocalDateTime;
import java.util.UUID;
import lombok.Getter;

@Getter
public class Snack {

  private String snackId;
  private String appointmentId;
  private String userId;
  private Double longitude;
  private Double latitude;
  private String snackImageUrl;
  private String snackSoundUrl;
  private LocalDateTime createdAt;

  public Snack(String snackId, String appointmentId, String userId, Double longitude,
      Double latitude,
      String snackImageUrl, String snackSoundUrl, LocalDateTime createdAt) {
    this.snackId = snackId;
    this.appointmentId = appointmentId;
    this.userId = userId;
    this.longitude = longitude;
    this.latitude = latitude;
    this.snackImageUrl = snackImageUrl;
    this.snackSoundUrl = snackSoundUrl;
    this.createdAt = createdAt;
  }

  public static Snack createNewSnack(String appointmentId, String userId, Double longitude,
      Double latitude, String snackImageUrl, String snackSoundUrl) {
    return new Snack(
        UUID.randomUUID().toString(),
        appointmentId,
        userId,
        longitude,
        latitude,
        snackImageUrl,
        snackSoundUrl,
        null
    );
  }
}

File: Witch/app/src/main/java/com/ssafy/witch/data/model/dto/Snack.kt
package com.ssafy.witch.data.model.dto

data class Snack(
    val snackId: String,
    val user: User,
    val createdAt : String,
    val snackImageUrl: String,
    val snackSoundUrl: String,
    val latitude: Double,
    val longitude: Double,
    val snackImageObjectKey: String?,
    val snackSoundObjectKey: String?,
    val appointmentId: String?
){
    constructor(latitude: Double, longitude: Double, snackImageObjectKey: String?, snackSoundObjectKey: String?):this("", User(), "", "", "", latitude, longitude, snackImageObjectKey, snackSoundObjectKey,"")
    constructor(nickname: String,latitude: Double, longitude: Double, snackImageUrl: String, snackSoundUrl: String): this("", User(nickname), "", snackImageUrl, snackSoundUrl, latitude, longitude, "", "","")
    constructor() : this("", User(), "", "", "", 0.0, 0.0, "", "","")
}
